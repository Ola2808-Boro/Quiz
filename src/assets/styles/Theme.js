export const theme = {
  colors: {
    backgroundColor: "#F5F7FB",
    text: "#293264",
    buttonColor: "#F5F7FB",
    circleBottom: "#DEEBF8",
    circleUp: "#FFFAD1",
    buttonBg: "#4D5B9E",
    borderAnswer: "#4D5B9E",
  },

  fontSize: {
    huge: "70px",
    xlgrand: "44px",
    lgrand: "40px",
    grand: "37px",
    xxxxl: "34px",
    xxxl: "33px",
    xxl: "30 px",
    xl: "26px",
    mediumxl: "24px",
    l: "22px",
    ml: "18px",
    m: "14px",
    s: "10px",
  },
};
