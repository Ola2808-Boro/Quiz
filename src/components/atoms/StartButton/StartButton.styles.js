import styled from "styled-components";
import { device } from "helpers/breakpoints";

export const Button = styled.button`
  font-size: ${({ theme }) => theme.fontSize.l};
  background-color: ${({ theme }) => theme.colors.buttonBg};
  letter-spacing: 1px;
  border-radius: 10px;
  color: white;
  border: none;
  padding: 10px 20px;
  width: 50%;

  @media ${device.mobileS} {
    width: 70%;
  }

  @media ${device.mobileL} {
    font-size: ${({ theme }) => theme.fontSize.mediumxl};
  }
  @media ${device.tablet} {
    font-size: ${({ theme }) => theme.fontSize.xxl};
    width: 50%;
  }
  @media ${device.laptop} {
    width: 60%;
    font-size: ${({ theme }) => theme.fontSize.xxxl};
  }
  @media ${device.laptopL} {
    width: 40%;
  }
  @media ${device.desktop} {
    width: 30%;
    font-size: ${({ theme }) => theme.fontSize.grand};
  }
`;
