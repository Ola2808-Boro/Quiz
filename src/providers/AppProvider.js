import React from 'react';
import { GlobalStyle } from 'assets/styles/GlobalsStyle'
import { ThemeProvider } from 'styled-components';
import 'assets/styles/fonts.css'
import {theme} from 'assets/styles/Theme'


const AppProviders = ({ children }) => {
    return (
 
          <ThemeProvider theme={theme}>      
                <GlobalStyle />
                {children}
          </ThemeProvider>
 
    );
  };
  
  export default AppProviders;