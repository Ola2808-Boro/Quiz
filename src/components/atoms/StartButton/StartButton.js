import React, { useContext } from "react";
import { Button } from "./StartButton.styles";
import { optionContext } from "providers/OptionProvider";
const StartButton = () => {
  const { handleClick } = useContext(optionContext);

  return <Button onClick={handleClick}>Start quiz</Button>;
};

export default StartButton;
