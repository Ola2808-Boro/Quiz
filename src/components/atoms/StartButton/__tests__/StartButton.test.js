import React from "react";
import "jest-styled-components";
import { Button } from "components/atoms/StartButton/StartButton.styles";
import AppProvider from "providers/AppProvider";
import StartButton from "../StartButton";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
describe("Test start button", () => {
  it("default style", () => {
    const tree = renderer
      .create(
        <AppProvider>
          <Button />
        </AppProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("test button content", () => {
    render(
      <AppProvider>
        <StartButton />
      </AppProvider>
    );
    const button = screen.getByRole("button", { name: /start quiz/i });
  });
});
