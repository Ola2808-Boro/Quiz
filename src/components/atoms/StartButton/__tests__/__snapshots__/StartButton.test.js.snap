// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test start button default style 1`] = `
<button
  className="eitdAa"
/>
`;
